src/meta/meta_ckpt.c

```diff
@@ -106,25 +106,17 @@ __wt_meta_checkpoint_clear(WT_SESSION_IMPL *session, const char *fname)
 static int
 __ckpt_set(WT_SESSION_IMPL *session, const char *fname, const char *v)
 {
+    WT_DECL_ITEM(tmp);
     WT_DECL_RET;
-    char *config, *newcfg;
-    const char *cfg[3];
 
-    config = newcfg = NULL;
-
-    /* Retrieve the metadata for this file. */
-    WT_ERR(__wt_metadata_search(session, fname, &config));
-
-    /* Replace the checkpoint entry. */
-    cfg[0] = config;
-    cfg[1] = v == NULL ? "checkpoint=()" : v;
-    cfg[2] = NULL;
-    WT_ERR(__wt_config_collapse(session, cfg, &newcfg));
-    WT_ERR(__wt_metadata_update(session, fname, newcfg));
+    WT_ERR(__wt_scr_alloc(session, 0, &tmp));
+    WT_ASSERT(session, strcmp(session->dhandle->name, fname) == 0);
+    WT_ERR(__wt_buf_fmt(session, tmp, "%s", session->dhandle->meta_base));
+    WT_ERR(__wt_buf_catfmt(session, tmp, ",%s", v == NULL ? "checkpoint=()," : v));
```

Comment:
There's no need for two calls, just do `"%s,%s"`.

Reply:
Fixed.